app/services/search/classified_listing.rb

```diff
@@ -3,10 +3,38 @@ class ClassifiedListing < Base
     INDEX_NAME = "classified_listings_#{Rails.env}".freeze
     INDEX_ALIAS = "classified_listings_#{Rails.env}_alias".freeze
     MAPPINGS = JSON.parse(File.read("config/elasticsearch/mappings/classified_listings.json"), symbolize_names: true).freeze
+    DEFAULT_PAGE = 0
+    DEFAULT_PER_PAGE = 75
 
     class << self
+      def search_documents(params:)
+        set_query_size(params)
+        query_hash = Search::QueryBuilders::ClassifiedListing.new(params).as_hash
+
+        results = search(body: query_hash)
+        hits = results.dig("hits", "hits").map { |cl_doc| cl_doc.dig("_source") }
+        paginate_hits(hits, params)
+      end
+
       private
 
+      def search(body:)
+        SearchClient.search(index: INDEX_ALIAS, body: body)
```

Comment:
we need to wrap this in

```ruby
request do
end
```

as we do in https://github.com/thepracticaldev/dev.to/blob/df93e8d41ef24e474fcb088070623417d8eef348/app/services/search/tag.rb#L9

Reply:
I would say lets just leave this since this will be going out soon https://github.com/thepracticaldev/dev.to/pull/6307 and then it will be taken care of